File: legacy/pasarguardbot/app/models/panel/tools.py
"""Admin panel DTOs: system status, backups and bulk volume/time increase."""

from pydantic import BaseModel, Field

from app.models.panel.common import PanelRequest, PanelResponse
from app.models.panel.services import PanelPanelOption


class PanelSystemMetrics(BaseModel):
    cpu_percent: float = 0
    cpu_cores: int = 0
    ram_percent: float = 0
    ram_used: int = 0
    ram_total: int = 0
    disk_percent: float = 0
    disk_used: int = 0
    disk_total: int = 0
    python: str | None = None
    platform: str | None = None


class PanelVersions(BaseModel):
    app: str | None = None
    telethon: str | None = None
    telethon_layer: str | None = None
    fastapi: str | None = None
    pasarguard: str | None = None
    database: str | None = None


class PanelScheduledJob(BaseModel):
    id: str
    last_run: str | None = None
    next_run: str | None = None


class PanelToolsResponse(PanelResponse):
    metrics: PanelSystemMetrics = Field(default_factory=PanelSystemMetrics)
    versions: PanelVersions = Field(default_factory=PanelVersions)
    jobs: list[PanelScheduledJob] = Field(default_factory=list)
    panels: list[PanelPanelOption] = Field(default_factory=list)
    backup_supported: bool = False


class PanelBulkIncreaseRequest(PanelRequest):
    """``panel`` is a panel code, or ``all`` for every panel."""

    panel: str = "all"
    volume_gb: float | None = Field(None, gt=0)
    days: int | None = Field(None, gt=0)
    confirm: bool = False

File: legacy/pasarguardbot/frontend/src/types/panel/tools.ts
/** Mirrors app/models/panel/tools.py */
import type { PanelAuthRequest, PanelEnvelope, PanelOption } from "./common";

export interface PanelSystemMetrics {
  cpu_percent: number;
  cpu_cores: number;
  ram_percent: number;
  ram_used: number;
  ram_total: number;
  disk_percent: number;
  disk_used: number;
  disk_total: number;
  python?: string | null;
  platform?: string | null;
}

export interface PanelVersions {
  app?: string | null;
  telethon?: string | null;
  telethon_layer?: string | null;
  fastapi?: string | null;
  pasarguard?: string | null;
  database?: string | null;
}

export interface PanelScheduledJob {
  id: string;
  last_run?: string | null;
  next_run?: string | null;
}

export interface PanelToolsResponse extends PanelEnvelope {
  metrics: PanelSystemMetrics;
  versions: PanelVersions;
  jobs: PanelScheduledJob[];
  panels: PanelOption[];
  backup_supported: boolean;
}

/** `panel` is a panel code, or "all" for every panel. */
export interface PanelBulkIncreaseRequest extends PanelAuthRequest {
  panel?: string;
  volume_gb?: number | null;
  days?: number | null;
  confirm: boolean;
}
